lib/impress.state.js

```diff
@@ -212,24 +209,23 @@ impress.state.mixin = (application) => {
   };
 
   application.state.on('change', (data, isTarget) => {
-    if (isTarget) {
-      let key, item, value;
-      const delta = data.delta;
-      for (key in delta) {
-        item = delta[key];
-        if (item.type === 'inc') {
-          value = api.common.getByPath(state.data, key);
-          value += item.value;
-          api.common.setByPath(state.data, key, value);
-        } else if (item.type === 'set') {
-          value = item.value;
-          api.common.setByPath(state.data, key, value);
-        } else if (item.type === 'del') {
-          value = undefined;
-          api.common.deleteByPath(state.data, key, value);
-        }
-        application.state.change(key, value, true);
+    if (!isTarget) return;
+    let key, item, value;
+    const delta = data.delta;
+    for (key in delta) {
+      item = delta[key];
+      if (item.type === 'inc') {
+        value = api.common.getByPath(state.data, key);
+        value += item.value;
+        api.common.setByPath(state.data, key, value);
+      } else if (item.type === 'set') {
+        value = item.value;
+        api.common.setByPath(state.data, key, value);
+      } else if (item.type === 'del') {
```

Comment:
Recurring code pattern. It should be viable to extract transformer functions and leave only generic code here.